contracts/test/EscrowEchidnaTest.sol

```diff
@@ -0,0 +1,80 @@
+
+    function deploy() external returns (address addr) {
+        bytes memory bytecode = daiBytecode;
+        assembly {
+            addr := create(0, add(bytecode, 0x20), mload(bytecode))
+        }
+    }
+}
+
+/// @dev Escrow Echidna Testing
+contract EscrowEchidnaTest {
+
+    address internal dai;
+    L1Escrow internal escrow;
+    OVM_L1CrossDomainMessenger internal messenger1;
+    Lib_AddressManager internal manager;
+    OVM_L2CrossDomainMessenger internal messenger2;
+    L2Gateway internal gate2;
+    L1Gateway internal gate1;
+    Dai internal oDai;
+    Create internal create;
+
+    uint256 internal constant WAD = 10**18;
+    uint256 internal constant MAX_SUPPLY = 10**15 * WAD;
+
+    /// @dev
+    constructor () {
+        create = new Create();
+        dai = create.deploy();
+        oDai = new Dai();
+        escrow = new L1Escrow();
+        messenger1 = new OVM_L1CrossDomainMessenger();
+        manager = new Lib_AddressManager();
+        messenger2 = new OVM_L2CrossDomainMessenger(address(manager));
+        gate2 = new L2Gateway(address(messenger2), address(oDai));
+        gate1 = new L1Gateway(iOVM_ERC20(dai), address(gate2), address(messenger1), address(escrow));
+    }
+
+    // --- Math ---
+    function add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x + y;
+        assert (z >= x); // check if there is an addition overflow
+    }
+    function sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x - y;
+        assert (z <= x); // check if there is a subtraction overflow
+    }
+
+    /// @dev
+    function bridge(uint256 wad) public {
+        uint256 supply = DaiLike(dai).totalSupply();
+        uint256 holderBalance = DaiLike(dai).balanceOf(address(this));
+        L1Escrow(escrow).approve(dai, address(this), type(uint256).max);
+        wad = 1 + wad % sub(MAX_SUPPLY, supply);
+        DaiLike(dai).mint(address(this), wad);
+        assert(DaiLike(dai).balanceOf(address(this)) == add(holderBalance, wad));
+        gate1.deposit(wad);
+        assert(DaiLike(dai).balanceOf(address(escrow)) >= oDai.totalSupply());
```

Comment:
`oDai` never gets minted in this test

Reply:
+1 need to complete the flow; also, seems like we should have a fuzzing function that goes the other way as well? (L2 --> L1)

Reply:
Just added the MockMessenger to enable oDai minting on `gate1.deposit` calls.